src/WidgetBase.ts

```diff
@@ -409,11 +454,22 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 			const render = beforeRenders.reduce((render, beforeRenderFunction) => {
 				return beforeRenderFunction.call(this, render, this._properties, this._children);
 			}, this.render.bind(this));
+
+			this._requiredNodes.forEach((element, key) => {
+				if (!this._nodeMap.has(key)) {
+					throw new Error(`Required node ${key} not found`);
+				}
+			});
+			this._requiredNodes.clear();
+
 			let dNode = render();
 			const afterRenders = this.getDecorator('afterRender');
 			afterRenders.forEach((afterRenderFunction: Function) => {
 				dNode = afterRenderFunction.call(this, dNode);
 			});
+
+			this._nodeMap.clear();
```

Comment:
Won't this be a problem if, on a re-render, the nodes in `_requireNodes` aren't created or updated and are then not passed into `_nodeMap` through `_setNode` which will throw the error above?